server/src/main/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -167,6 +174,42 @@ public static long sizeOfRelocatingShards(
     @Override
     public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
         ClusterInfo clusterInfo = allocation.clusterInfo();
+
+        /*
+         The following block enables allocation for remote shards within safeguard limits of the filecache.
+         */
+        if (REMOTE_CAPABLE.equals(getNodePool(node)) && REMOTE_CAPABLE.equals(getShardPool(shardRouting, allocation))) {
```

Comment:
It looks to me we have now 2 independent deciders: for REMOTE_CAPABLE and non-REMOTE_CAPABLE. Should we split then as such?

Reply:
@reta I think with the next phase (remote indices), we can possibly break it down into a `FileCacheDecider`. I think its good here for now, since it's under a related operation.

Reply:
Sure, could you please link the issue so won't forget it? Thank you.

Reply:
#8535